File: frontend/src/api/complaintAnalysis.ts
import { apiFetch } from './httpClient'

export type ComplaintTextAnalysisResponse = {
  categories: string[]
  sentiment: string
  severity: string
  location: string | null
  summary: string
  keywords: string[]
}

export async function analyzeComplaintText(
  token: string,
  text: string,
): Promise<ComplaintTextAnalysisResponse> {
  return apiFetch<ComplaintTextAnalysisResponse>('/complaints/analyze-text', {
    method: 'POST',
    token,
    body: JSON.stringify({ text }),
  })
}

File: backend/app/schemas/complaint_analysis.py
from pydantic import BaseModel, Field


class ComplaintTextAnalysisRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=5000)


class ComplaintTextAnalysisResponse(BaseModel):
    categories: list[str]
    sentiment: str
    severity: str
    location: str | None
    summary: str
    keywords: list[str]
